src/oca/go/src/goca/virtualrouter.go

```diff
@@ -0,0 +1,183 @@
+	return &VirtualRouter{ID: id}
+}
+
+// NewVirtualRouterFromName finds a virtual router object by name. It connects to
+// OpenNebula to retrieve the pool, but doesn't perform the Info() call to
+// retrieve the attributes of the virtual router.
+func NewVirtualRouterFromName(name string) (*VirtualRouter, error) {
+	vrouterPool, err := NewVirtualRouterPool()
+	if err != nil {
+		return nil, err
+	}
+
+	id, err := vrouterPool.GetIDFromName(name, "/VROUTER_POOL/VROUTER")
+	if err != nil {
+		return nil, err
+	}
+
+	return NewVirtualRouter(id), nil
+}
+
+// CreateVirtualRouter allocates a new virtual router. It returns the new Virtual Router ID
+// * tpl: template of the marketplace
+func CreateVirtualRouter(tpl string) (uint, error) {
+	response, err := client.Call("one.vrouter.allocate", tpl)
+	if err != nil {
+		return 0, err
+	}
+
+	return uint(response.BodyInt()), nil
+}
+
+// Info connects to OpenNebula and fetches the information of the VirtualRouter
+func (vr *VirtualRouter) Info() error {
+	response, err := client.Call("one.vrouter.info", vr.ID)
+	vr.body = response.Body()
+	return err
+}
+
+// Update will modify the virtual router. If appendVirtualRouter is 0, it will
+// replace the whole virtual router. If its 1, it will merge.
+func (vr *VirtualRouter) Update(tpl string, appendVirtualRouter int) error {
+	_, err := client.Call("one.vrouter.update", vr.ID, tpl, appendVirtualRouter)
+	return err
+}
+
+// Chown changes the owner/group of a virtual router. If uid or gid is -1 it will not
+// change
+func (vr *VirtualRouter) Chown(uid, gid int) error {
+	_, err := client.Call("one.vrouter.chown", vr.ID, uid, gid)
+	return err
+}
+
+// Chmod changes the permissions of a virtual router. If any perm is -1 it will not
+// change
+func (vr *VirtualRouter) Chmod(uu, um, ua, gu, gm, ga, ou, om, oa int) error {
+	_, err := client.Call("one.vrouter.chmod", vr.ID, uu, um, ua, gu, gm, ga, ou, om, oa)
+	return err
+}
+
+// Rename changes the name of virtual router
+func (vr *VirtualRouter) Rename(newName string) error {
+	_, err := client.Call("one.vrouter.rename", vr.ID, newName)
+	return err
+}
+
+// Delete will remove the virtual router from OpenNebula.
+func (vr *VirtualRouter) Delete() error {
+	_, err := client.Call("one.vrouter.delete", vr.ID)
+	return err
+}
+
+// Instantiate will instantiate the virtual router. It returns the ID of the new VM
+// * number: Number of VMs to instantiate.
+// * tplid: VM Template id to instantiate.
+// * name: Name for the VM instances. If it is an empty string OpenNebula will set a default name. Wildcard %i can be used.
+// * pending: False to create the VM on pending (default), True to create it on hold.
+// * extra: A string containing an extra template to be merged with the one being instantiated. It can be empty. Syntax can be the usual attribute=value or XML.
+func (vr *VirtualRouter) Instantiate(number, tplid int, name string, pending bool, extra string) (uint, error) {
```

Comment:
I think it would be better to rename the pending parameter by hold. It seem weird to me to call it pending when it need to be false for the VM to be instantiated in pending state.
```suggestion
func (vr *VirtualRouter) Instantiate(number, tplid int, name string, hold bool, extra string) (uint, error) {
```

Reply:
indeed!

Reply:
yes. fixed in another commit